module/Kbuild.in

```diff
@@ -39,6 +39,15 @@ ifneq ($(KBUILD_EXTMOD),)
 @CONFIG_QAT_TRUE@KBUILD_EXTRA_SYMBOLS += @QAT_SYMBOLS@
 endif
 
+# Woe betide ye who override this, for yours is the kingdom of mysterious segfaults
+# absolutely everywhere.
+#
+# Free warning to anyone considering it - -march=native or the like later in the line
+# will undo -mgeneral-regs-only, and gcc's -O2 starting in 12 does autovectorizing.
+#
+# Good luck.
+ZFS_MODULE_CFLAGS += -mgeneral-regs-only -fno-tree-vectorize
```

Comment:
The Linux kernel really should handle this for us by disabling all of the vector instructions such that there is no need to pass either -mgeneral-regs-only or -fno-tree-vectorize.

`-mgeneral-regs-only` is a convenience flag meant to avoid requiring people to know about all various isa extensions. It is not available on all architectures and it will break the build on x86_64 on GCC versions older than gcc 7.

`-fno-tree-vectorize` is only specified in Linux's `arch/csky/Makefile` and no where else in Linux, since it really should not be necessary. That said, I am fine with passing `-fno-tree-vectorize` here.

Reply:
I didn't just add it for fun, no.

As the comment above explains, if the user forcibly overrides -march later in the line, then these are overridden as well, and boom goes the dynamite.

Reply:
Alright. -mgeneral-regs-only is not backward compatible with older compilers so unless we write an autotools check for it, we probably should drop it in favor of only doing -fno-tree-vectorize.

Reply:
I was going to write an autoconf check, yes.

I think -fno-tree-vectorize might also need a check of that nature, but it's been a while since I looked, so I'll check when I get back to this.